amigo-lib/src/main/java/me/ele/amigo/utils/ComponentUtils.java

```diff
@@ -0,0 +1,186 @@
+                if (appReceivers != null && appReceivers.length > 0) {
+                    for (ActivityInfo appReceiver : appReceivers) {
+                        if (newAppReceiver.name.equals(appReceiver.name)) {
+                            isNew = false;
+                            break;
+                        }
+                    }
+                }
+
+                if (isNew) {
+                    Log.e(TAG, "newReceiver-->" + newAppReceiver);
+                    addedReceivers.add(newAppReceiver);
+                }
+            }
+        }
+        return addedReceivers;
+    }
+
+
+    public static void registerNewReceivers(Context context) {
+        try {
+            List<ActivityInfo> addedReceivers = getNewAddedReceivers(context);
+            for (ActivityInfo addedReceiver : addedReceivers) {
+                List<IntentFilter> filters = getReceiverIntentFilter(context, addedReceiver);
+                for (IntentFilter filter : filters) {
+                    BroadcastReceiver receiver = (BroadcastReceiver) ComponentUtils.class.getClassLoader().loadClass(addedReceiver.name).newInstance();
+                    context.registerReceiver(receiver, filter);
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static List<IntentFilter> getReceiverIntentFilter(Context context, ActivityInfo receiverInfo) throws InvocationTargetException, NoSuchMethodException, ClassNotFoundException, InstantiationException, IllegalAccessException {
+        List<Object> receivers = getReceivers(context);
+        Object data = null;
+        for (Object receiver : receivers) {
+            ActivityInfo info = (ActivityInfo) FieldUtils.readField(receiver, "info");
+            if (info.name.equals(receiverInfo.name)) {
+                data = receiver;
+                break;
+            }
+        }
+        return (List<IntentFilter>) FieldUtils.readField(data, "intents");
+    }
+
+    private static List<Object> receivers;
+
+    private static List<Object> getReceivers(Context context) throws IllegalAccessException, NoSuchMethodException, InvocationTargetException, InstantiationException, ClassNotFoundException {
+        if (receivers != null) {
+            return receivers;
+        }
+        Class sPackageParserClass = Class.forName("android.content.pm.PackageParser");
+        Object mPackageParser;
+        Object mPackage;
+
+        File file = getHotFixApk(context);
+        if(file == null ) {
+            return Collections.EMPTY_LIST;
+        }
+
+        try {
+            mPackageParser = sPackageParserClass.newInstance();
+            mPackage = MethodUtils.invokeMethod(mPackageParser, "parsePackage", file, 0);
+        } catch (Exception e) {
+            e.printStackTrace();
+
+            mPackageParser = sPackageParserClass.getDeclaredConstructor(String.class).newInstance(file.getPath());
```

Comment:
I think use build version rather than exception is better for this kind of situation